mmv1/third_party/terraform/tests/resource_network_services_grpc_route_test.go.erb

```diff
@@ -0,0 +1,97 @@
+<% autogen_exception -%>
+package google
+<% unless version == 'ga' -%>
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccNetworkServicesGrpcRoute_update(t *testing.T) {
+	t.Parallel()
+
+	grpcRouteName := fmt.Sprintf("tf-test-grpc-route-%s", RandString(t, 10))
+
+	VcrTest(t, resource.TestCase{
+		PreCheck:     func() { AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy: testAccCheckNetworkServicesGrpcRouteDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+        Config: testAccNetworkServicesGrpcRoute_basic(grpcRouteName),
+			},
+			{
+				ResourceName:      "google_network_services_grpc_route.foobar",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccNetworkServicesGrpcRoute_update(grpcRouteName),
+			},
+			{
+				ResourceName:      "google_network_services_grpc_route.foobar",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccNetworkServicesGrpcRoute_basic(grpcRouteName string) string {
+	return fmt.Sprintf(`
+  resource "google_network_services_grpc_route" "foobar" {
+    name                   = "%s"
+    labels                 = {
+      foo = "bar"
+    }
+    description             = "my description"
+    hostnames               = ["example"]
+    rules                   {
+      matches {
+        headers {
+          key = "key"
+          value = "value"
+        }
+      }
+      action {
+        retry_policy {
+            retry_conditions = ["server_error"]
+            num_retries = 1
+        }
+      }
+    }
+  }
+`, grpcRouteName)
+}
+
+func testAccNetworkServicesGrpcRoute_update(grpcRouteName string) string {
+	return fmt.Sprintf(`
+  resource "google_network_services_grpc_route" "foobar" {
+    name                   = "%s"
+    labels                 = {
+      foo = "bar"
+    }
+    description             = "updated description"
+    hostnames               = ["example"]
+    rules                   {
```

Comment:
Can you update some of the more complicated fields in this test? Things like, adding more rules, removing rules, setting empty rules, changing fields within a rule

Reply:
Done!